### PR TITLE
 worker: Unify retry-behavior of artefact file uploads with regular API calls

### PR DESCRIPTION
**Pull request by Martchus**
Based on #4063, see particular commit messages and https://progress.opensuse.org/issues/94531

### CODE REVIEW

**Review by okurz — APPROVED** [review]

File: lib/OpenQA/Worker/Job.pm [hunk 1]
```diff
@@ -981,22 +984,19 @@ sub _upload_asset {
...
+        'upload_chunk.prepare' => sub ($upload, $pieces, @args) {
             log_info("$filename: " . $pieces->size() . " chunks",   channels => \@channels_worker_only, default => 1);
             log_info("$filename: chunks of $chunk_size bytes each", channels => \@channels_worker_only, default => 1);
         });
     my $t_start;
     $ua->upload->on('upload_chunk.start' => sub { $t_start = time() });
     $ua->upload->on(
-        'upload_chunk.finish' => sub {
-            my ($self, $piece) = @_;
+        'upload_chunk.finish' => sub ($upload, $piece, @args) {
```
Comment (kraih) [inline lib/OpenQA/Worker/Job.pm:999]: This event doesn't actually have more arguments, what is `@args` supposed to do here?

File: lib/OpenQA/Worker/Job.pm [hunk 2]
```diff
@@ -981,22 +984,19 @@ sub _upload_asset {
     log_info("Uploading $filename using multiple chunks", channels => \@channels_worker_only, default => 1);
 
     $ua->upload->once(
-        'upload_local.prepare' => sub {
-            my ($self) = @_;
+        'upload_local.prepare' => sub ($upload, @args) {
```
Comment (kraih) [inline lib/OpenQA/Worker/Job.pm:987]: The `@args` is not necessary.

File: lib/OpenQA/Worker/Job.pm [hunk 3]
```diff
@@ -1010,21 +1010,19 @@ sub _upload_asset {
...
         }
-        _log_upload_error($filename, $res);
+        $self->_log_upload_error($filename, $tx);
     };
     $ua->upload->on('upload_local.response' => $response_cb);
     $ua->upload->on('upload_chunk.response' => $response_cb);
     $ua->upload->on(
-        'upload_chunk.fail' => sub {
-            my ($self, $res, $chunk) = @_;
+        'upload_chunk.fail' => sub ($upload, $res, $chunk, @args) {
```
Comment (kraih) [inline lib/OpenQA/Worker/Job.pm:1025]: The @args is not necessary.

File: lib/OpenQA/Worker/Job.pm [hunk 4]
```diff
@@ -1010,21 +1010,19 @@ sub _upload_asset {
             );
         });
 
-    my $response_cb = sub {
-        my ($self, $res) = @_;
-        if ($res->res->is_server_error) {
-            log_error($res->res->json->{error}, channels => \@channels_both, default => 1)
-              if $res->res->json && $res->res->json->{error};
+    my $response_cb = sub ($upload, $tx, @args) {
```
Comment (kraih) [inline lib/OpenQA/Worker/Job.pm:1013]: The `@args` is not necessary.

File: lib/OpenQA/Worker/Job.pm [hunk 5]
```diff
@@ -981,22 +984,19 @@ sub _upload_asset {
...
-        'upload_local.prepare' => sub {
-            my ($self) = @_;
+        'upload_local.prepare' => sub ($upload, @args) {
             log_info("$filename: local upload (no chunks needed)", channels => \@channels_worker_only, default => 1);
             chmod 0644, $file;
         });
     $ua->upload->once(
-        'upload_chunk.prepare' => sub {
-            my ($self, $pieces) = @_;
+        'upload_chunk.prepare' => sub ($upload, $pieces, @args) {
```
Comment (kraih) [inline lib/OpenQA/Worker/Job.pm:992]: The `@args` is not necessary.

**Review by kraih — APPROVED** [review]

**Pull request merged**